File: api/src/main/java/com/api/app/entity/GoodsPriceHist.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("GoodsPriceHist")
@Getter
@Setter
public class GoodsPriceHist implements Serializable {
    private static final long serialVersionUID = 4456789012345678901L;

    @Schema(description = "상품번호")
    private String goodsNo;

    @Schema(description = "시작일")
    private LocalDateTime startDateTime;

    @Schema(description = "종료일")
    private LocalDateTime endDateTime;

    @Schema(description = "판매가")
    private Long salePrice;

    @Schema(description = "공급원가")
    private Long supplyPrice;
}

File: api/src/main/kotlin/com/api/app/entity/GoodsPriceHist.kt
package com.api.app.entity

import jakarta.persistence.*
import java.io.Serializable
import java.time.LocalDateTime

@Embeddable
data class GoodsPriceHistId(
    @Column(name = "GOODS_NO", length = 15) val goodsNo: String = "",
    @Column(name = "START_DATE_TIME") val startDateTime: LocalDateTime = LocalDateTime.now()
) : Serializable

@Entity
@Table(name = "GOODS_PRICE_HIST")
class GoodsPriceHist : SystemEntity() {

    @EmbeddedId
    var id: GoodsPriceHistId = GoodsPriceHistId()

    @Column(name = "END_DATE_TIME")
    var endDateTime: LocalDateTime = LocalDateTime.of(9999, 12, 31, 23, 59, 59)

    @Column(name = "SALE_PRICE", nullable = false, columnDefinition = "numeric")
    var salePrice: Long = 0

    @Column(name = "SUPPLY_PRICE", nullable = false, columnDefinition = "numeric")
    var supplyPrice: Long = 0

    // Convenience accessors
    val goodsNo: String get() = id.goodsNo
    val startDateTime: LocalDateTime get() = id.startDateTime
}
